pkg/envoy/types.go

```diff
@@ -24,20 +23,16 @@ func (t TypeURI) String() string {
 
 // Short returns an abbreviated version of the TypeURI, which is easier to spot in logs and metrics.
 func (t TypeURI) Short() string {
-	if t == TypeWildcard {
-		return Wildcard
-	}
-	return utils.GetLastChunkOfSlashed(t.String())
+	return XDSShortURINames[t]
 }
 
 // IsWildcard returns true when the TypeURI is the one requesting ALL discovery types.
 func (t TypeURI) IsWildcard() bool {
-	return t == TypeWildcard
+	return t == TypeEmptyURI
 }
 
 // ValidURI defines valid URIs
 var ValidURI = map[string]TypeURI{
-	string(TypeWildcard):           TypeWildcard,
```

Comment:
Removing empty typeURI results in error logs, which is not correct per these changes:
```
{"level":"error","component":"envoy/ads","time":"2021-04-30T16:44:57Z","file":"stream.go:174","message":"Proxy SerialNumber=136867002654723736450590910962478278124 PodUID=a9d0fb42-e2ce-4596-a3aa-92631baeb4ce: Unknown/Unsupported URI: "}
```

Reply:
I will instead, allow `emptyUrl` as valid, but add a check just right after to skip empty url with debug log instead